### ISSUE TITLE
Unable to build demo project with latest version of rust

### ISSUE THREAD

**Issue by okayplanet** [rust-lang/rust#102937]
I tried this code:

```rust
cargo run --release
```

I expected to see this happen: I expected the new project that I created to build, from following https://guide.nannou.cc/getting_started/create_a_project.html

Instead, this happened: 
<img width="887" alt="Screen Shot 2022-10-11 at 4 49 40 PM" src="https://user-images.githubusercontent.com/3144575/195205015-1ed8f47f-7eb9-4e77-ae14-7f838c6a6b85.png">

### Meta

`rustc --version --verbose`:
```
rustc 1.64.0 (a55dd71d5 2022-09-19)
binary: rustc
commit-hash: a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52
commit-date: 2022-09-19
host: x86_64-apple-darwin
release: 1.64.0
LLVM version: 14.0.6
```

<details><summary>Backtrace</summary>
<p>

```
error[E0597]: `desc_set` does not live long enough
    --> /Users/brittcagnina/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-core-0.9.2/src/device/mod.rs:1792:26
     |
1792 |                     set: desc_set.raw_mut(),
     |                          ^^^^^^^^^^^^^^^^^^ borrowed value does not live long enough
...
1813 |     }
     |     -
     |     |
     |     `desc_set` dropped here while still borrowed
     |     borrow might be used here, when `write_map` is dropped and runs the `Drop` code for type `BTreeMap`
     |
     = note: values in a scope are dropped in the opposite order they are defined

For more information about this error, try `rustc --explain E0597`.
error: could not compile `wgpu-core` due to previous error
```

</p>
</details>

**Comment by ehuss** [comment]
Sorry for the trouble you've run into. There was a change to Rust to fix a soundness problem (#99413, #100544). That requires updating to a newer version of wgpu-core that does not have the problem. It looks like you should be able to update to the latest version of `nannou` to fix it.  In `Cargo.toml`, you can update it to `nannou = "0.18"` to pick up the new version.

**Comment by okayplanet** [comment]
Perfect! Works great now! Thanks
